Gems/XR/Code/Source/XRSystem.cpp

```diff
@@ -370,4 +373,148 @@ namespace XR
+            //
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 0, 0, 2, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 0, gridSize - 1, 1, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x1, 2, 0, 2, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x1, 1, gridSize - 1, 2, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate1x2, 0, 1, 1, gridSize - 2);
+        }
+        break;
+        case AZ::RHI::XRFoveatedLevel::Medium:
+        {
+            //  _______________________________________________
+            // |_4x2_|________________2x2________________|_4x2_|
+            // |     | 1x2 |                       | 1x2 |     |
+            // |     |_____|                       |_____|     |
+            // |     |                                   |     |
+            // | 2x2 |                1x1                | 2x2 |
+            // |     |                                   |     |
+            // |     |___________________________________|     |
+            // |___________|__________2x1__________|___________|
+            //
+            fillFunc(AZ::RHI::ShadingRate::Rate4x2, 0, 0, 1, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 1, 0, 3, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 0, 1, 1, gridSize - 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 1, gridSize - 1, 1, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate1x2, 1, 1, 1, 2);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x1, 2, gridSize - 1, 2, 1);
+        }
+        break;
+        case AZ::RHI::XRFoveatedLevel::High:
+        {
+            //  _______________________________________________
+            // |_4x4_|____4x2____|____2x2____|____4x2____|_4x4_|
+            // | 2x4 |_2x2_|                       |_2x2_| 2x4 |
+            // |_____|_____|                       |_____|_____|
+            // |     |     |                       |     |     |
+            // | 2x2 | 1x2 |          1x1          | 1x2 | 2x2 |
+            // |     |     |                       |     |     |
+            // |_____|_____|_______________________|_____|_____|
+            // |_2x4_|____2x2____|____2x1____|____2x2____|_2x4_|
+            //
+            fillFunc(AZ::RHI::ShadingRate::Rate4x4, 0, 0, 1, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate4x2, 1, 0, 2, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x4, 0, 1, 1, 2);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x4, 0, gridSize - 1, 1, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 3, 0, 1, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 1, 1, 1, 2);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 0, 3, 1, 4);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x2, 1, gridSize - 1, 2, 1);
+            fillFunc(AZ::RHI::ShadingRate::Rate1x2, 1, 3, 1, 4);
+            fillFunc(AZ::RHI::ShadingRate::Rate2x1, 3, gridSize - 1, 2, 1);
+        }
+        break;
+        case AZ::RHI::XRFoveatedLevel::None:
+            break;
```

Comment:
Probably wants a comment here that says "Intentionally leave the rate grid with default values"